services/src/datasets/mod.rs

```diff
@@ -0,0 +1,5 @@
+pub mod access;
+pub mod in_memory;
+pub mod listing;
+pub mod postgres;
+pub mod storage;
```

Comment:
Why don't you just use the structs and traits? We should keep less hierarchical levels if possible.